assets/course-theme/course-theme.editor.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable @wordpress/no-global-event-listener */
+/**
+ * WordPress dependencies
+ */
+import { getQueryArgs, addQueryArgs } from '@wordpress/url';
+
+window.addEventListener( 'locationchange', redirectToCourseThemeOverride );
+window.addEventListener( 'popstate', redirectToCourseThemeOverride );
+
+/**
+ * Reload the page when opening or closing a course theme template, to ensure the active theme styles are not loaded.
+ */
+function redirectToCourseThemeOverride() {
+	const query = getQueryArgs( document.location );
+	const isCourseThemeDocument =
+		query.postId && query.postId.match( /sensei-course-theme/ );
+
+	const isCourseThemeActive = query.learn;
+
+	query.learn = isCourseThemeDocument ? '1' : undefined;
+	if ( !! query.learn !== !! isCourseThemeActive ) {
+		const url = addQueryArgs( document.location.path, query );
+		document.location.replace( url );
+	}
+}
+
+/**
+ * Monkey-patch history.pushState and replaceState to provide events for location change.
```

Comment:
Out of curiosity, what is the case that the `popstate` only didn't work?

Reply:
It's [not called](https://developer.mozilla.org/en-US/docs/Web/API/Window/popstate_event#the_history_stack) when using `pushState` or `replaceState`, just when the browser back/forward button is pressed unfortunately.

Reply:
Ah! 😞
I didn't remember that!